packages/database/test/query.test.ts

```diff
@@ -1399,6 +1399,38 @@ describe('Query Tests', () => {
     expect(Object.values(snap.val())).to.deep.equal([snap.val()[childOne.key]]);
   });
 
+  it('Ensure startAfter on key index works with overlapping listener', async () => {
+    const node = getRandomNode() as Reference;
+    const childOne = node.push();
+    const childTwo = node.push();
+    await childOne.set(1);
+    await childTwo.set(2);
```

Comment:
Do the tests still pass if you don't `await` here? I would like to confirm that we correctly apply local writes.

Reply:
They do still pass.